lib/form/validate.js

```diff
@@ -173,12 +173,28 @@ const buildValues = working => {
 };
 
 /*
+ * Checks for required fields before running validation
+ *
+ * @param {object} input - Input Type being tested
+ * @param {ValidationSettings} settings - All validation settings
+ *
+ * @returns {true|FormInputValues} - Returns `true` if there are no validation or required issues, {FormInputValues} with error messages as values if there are
+ */
+const required = (input, settings, validation) => {
+  if (input.target.required && (input.target.required === 'save') && input.target.value === '') {
```

Comment:
1. changed to `if (input.target.hasOwnProperty('required') && (input.target.required === 'save' || input.target.required === 'publish') && input.target.value === '') {`
2. checkboxes/radio - not sure what you mean. `value` would be blank if nothing was selected in either case.